### ISSUE TITLE
cabal2nix --shell doesn't include libraryToolDepends in environment 

### ISSUE THREAD

**Issue by mpickering** [NixOS/cabal2nix#331]
My workflow is usually

```
cabal2nix --shell . > shell.nix
nix-shell
cabal configure
cabal build
```

If the project uses `happy` or `alex` then these are not available in the shell environment as `libraryToolDepends` is not part of `propagatedBuildInputs`.

**Comment by peti** [comment]
I cannot re-produce this issue (on `master`): 

~~~
nix-shell "<nixpkgs>" --pure -A haskellPackages.ViennaRNA-bindings --run "type -pa c2hs"
/nix/store/pm7fq77pzyprwkp8vbw5cqqc9raimn58-c2hs-0.26.2-28-g8b79823/bin/c2hs
/nix/store/pm7fq77pzyprwkp8vbw5cqqc9raimn58-c2hs-0.26.2-28-g8b79823/bin/c2hs
~~~

The build tool `c2hs` shows up in `$PATH`. Generally speaking, the environment contains `systemBuildInputs`, not `propagatedBuildInputs`.

**Comment by mpickering** [comment]
https://gist.github.com/1f6d8fbf4b3d178c33c0a23b1f6f831a

I am starting with `cogent.cabal`. `shell.nix` is the generated `shell.nix` file with a lot of the dependencies removed (it still fails with the generated file). After running `nix-shell`, `alex` is not available on the command line.

**Comment by peti** [comment]
This is all very confusing. Anyway, I've committed a fix in 515c039b99195062298c67c0031bc3c8902a974e. I hope it works okay. :-)
